examples/src/bin/unbuffered-server.rs

```diff
@@ -0,0 +1,255 @@
+use std::error::Error;
+use std::fs::File;
+use std::io::{self, BufReader, Read, Write};
+use std::net::{TcpListener, TcpStream};
+use std::sync::Arc;
+
+use pki_types::{CertificateDer, PrivateKeyDer};
+use rustls::server::UnbufferedServerConnection;
+use rustls::unbuffered::{
+    AppDataRecord, ConnectionState, EncodeError, EncryptError, InsufficientSizeError,
+    UnbufferedStatus,
+};
+use rustls::ServerConfig;
+use rustls_pemfile::Item;
+
+const KB: usize = 1024;
+const INCOMING_TLS_BUFSIZ: usize = 16 * KB;
+const OUTGOING_TLS_INITIAL_BUFSIZ: usize = 0;
+const MAX_EARLY_DATA_SIZE: Option<u32> = Some(128);
+const MAX_FRAGMENT_SIZE: Option<usize> = None;
+
+const PORT: u16 = 1443;
+const MAX_ITERATIONS: usize = 30;
+const CERTFILE: &str = "localhost.pem";
+const PRIV_KEY_FILE: &str = "localhost-key.pem";
```

Comment:
could these be optionally sourced from env vars ? Having them hardcoded will make using it a little bit awkward.